src/redux/slices/userSlice.js

```diff
@@ -1,20 +1,288 @@
+    } catch (err) {
+      rejectWithValue(err);
+    }
+  }
+);
+
 const userSlice = createSlice({
-  name: 'users',
+  name: 'user',
   initialState: {
-    users: [],
     user: {},
+    error: '',
+    isLoading: false,
   },
-  reducers: {
-    testUser: (state) => {
-      state.user = { test: 'user' };
-    },
+  reducers: {},
+  extraReducers: (builder) => {
+    builder
+      .addCase(fetchUser.fulfilled, (state, { payload }) => {
+        state.user = payload;
+        state.isLoading = false;
+        state.error = '';
+      })
+      .addCase(fetchUser.pending, (state, { payload }) => {
+        state.user = {};
+        state.isLoading = true;
+        state.error = '';
+      })
+      .addCase(fetchUser.rejected, (state, action) => {
+        state.user = {};
+        state.isLoading = false;
+        state.error = action.payload.message;
+      })
+      .addCase(createNewUser.fulfilled, (state, { payload }) => {
+        state.user = payload;
+        state.isLoading = false;
+        state.error = '';
+      })
+      .addCase(createNewUser.pending, (state, { payload }) => {
+        state.user = {};
+        state.isLoading = true;
+        state.error = '';
+      })
+      .addCase(createNewUser.rejected, (state, action) => {
+        state.user = {};
+        state.isLoading = false;
+        state.error = action.payload.message;
+      })
+      .addCase(updateUser.fulfilled, (state, { payload }) => {
+        state.user = payload;
+        state.isLoading = false;
+        state.error = '';
+      })
+      .addCase(updateUser.pending, (state, { payload }) => {
+        state.user = {};
+        state.isLoading = true;
+        state.error = '';
+      })
+      .addCase(updateUser.rejected, (state, action) => {
+        state.user = {};
+        state.isLoading = false;
+        state.error = action.payload.message;
+      })
+      .addCase(updateLocation.fulfilled, (state, { payload }) => {
+        state.user = payload;
+        state.isLoading = false;
+        state.error = '';
+      })
+      .addCase(updateLocation.pending, (state, { payload }) => {
+        state.user = {};
+        state.isLoading = true;
+        state.error = '';
+      })
+      .addCase(updateLocation.rejected, (state, action) => {
+        state.user = {};
+        state.isLoading = false;
+        state.error = action.payload.message;
+      })
+      .addCase(deleteUser.fulfilled, (state, { payload }) => {
+        state.user = {};
+        state.isLoading = false;
+        state.error = '';
+      })
+      .addCase(deleteUser.pending, (state, { payload }) => {
+        state.user = {};
+        state.isLoading = true;
+        state.error = '';
+      })
+      .addCase(deleteUser.rejected, (state, action) => {
+        state.user = {};
+        state.isLoading = false;
+        state.error = action.payload.message;
+      })
+      .addCase(banUser.fulfilled, (state, { payload }) => {
+        state.user = {};
```

Comment:
Should user be set to empty just because someone is banned?

Reply:
good point: no...i set state.user = payload for both this + remove ban